sycl/include/CL/sycl/detail/cg.hpp

```diff
@@ -224,10 +223,37 @@ class HostKernel : public HostKernelBase {
   }
 
   template <class ArgT = KernelArgType>
-  typename std::enable_if<std::is_same<ArgT, nd_item<Dims>>::value>::type
+  typename std::enable_if<
+      std::is_same<ArgT, item<Dims, /*Offset=*/true>>::value>::type
   runOnHost(const NDRDescT &NDRDesc) {
-    // TODO add offset logic
+    size_t XYZ[3] = {0};
+    sycl::id<Dims> ID;
```

Comment:
Please, move `XYZ` and `ID` declaration after the first loop i.e. closer to their first use.

Reply:
Done.